### ISSUE TITLE
OOM Issue Apears in Speech to Text Task

### ISSUE THREAD

**Issue by zjw1990** [pytorch/fairseq#3096]
Hi, 

I was running the fariseq speech-to-text example and got OOM issue (either 1 or 4 v100 GPU). I tried the --empty-cache-freq flag. It still occurs OOM eventually although it takes longer to OOM. With 128 G mem

I also tried to limit the size of max-tokens but it seems the mem also slowly raised up. Is there anything I could avoid this?

To reproduce

#!/bin/bash
python train.py ${MUSTC_ROOT}/en-fr --train-subset train_st --valid-subset dev_st --save-dir ${ST_SAVE_DIR} \
    --max-tokens 10000 --task speech_to_text --criterion label_smoothed_cross_entropy --num-workers 0 \
    --max-update 200000 --arch s2t_transformer_s --optimizer adam --lr 2e-3 \
    --lr-scheduler inverse_sqrt --warmup-updates 10000 --clip-norm 10.0 --seed 1 --config-yaml ${MUSTC_ROOT}/en-fr/config_st.yaml \
    --empty-cache-freq 1 --report-accuracy --load-pretrained-encoder-from ${ASR_SAVE_DIR}/${ASR_CHECKPOINT_FILENAME}

Thanks.

**Comment by kahne** [comment]
With V100, you may try [`--fp16`](https://fairseq.readthedocs.io/en/latest/getting_started.html#training-with-half-precision-floating-point-fp16) to save GPU memory and speed-up training. When trying lower `--max-tokens`, did you find one that gets you through?
